internal/pkg/controllers/profilerecorder/profilerecorder.go

```diff
@@ -0,0 +1,252 @@
+package profilerecorder
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	seccompoperatorv1alpha1 "sigs.k8s.io/seccomp-operator/api/v1alpha1"
+	"sigs.k8s.io/seccomp-operator/internal/pkg/config"
+)
+
+const (
+	// default reconcile timeout.
+	reconcileTimeout = 1 * time.Minute
+
+	maxTraceAnnotations = 2
+
+	errGetClient         = "cannot get client connection"
+	errGetNode           = "cannot get node object"
+	errGetPod            = "cannot get pod"
+	errInvalidAnnotation = "invalid Annotation"
+
+	// traceAnnotation is the annotation on a Pod that triggers the
+	// oci-seccomp-bpf-hook to trace the syscalls of a Pod and created a
+	// seccomp profile.
+	traceAnnotation = "io.containers.trace-syscall"
```

Comment:
Did you use the CRI-O annotation patch to make this work? I'm thinking if we can choose another way to pass the data than the annotation…

Reply:
Yeah. For now this depends on the annotation patch. (For reference https://github.com/cri-o/cri-o/pull/4138)